### PR TITLE
Basic support for Espresso testing added

### PR DESCRIPTION
**Pull request by zugaldia**
Includes docs, base classes, and Gradle dependencies. Works both locally and on AWS Device Farm.

### CODE REVIEW

File: styles [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 19dde3cb1c36fe5f42dafc78c806fd7aca98126c
+Subproject commit 014e36833420c81b29bec816b479ea61e94821d4
```
Comment (jfirebaugh) [inline styles]: You probably didn't mean to change any of these submodule pins. Committing them accidentally is a common mistake.

Comment (bleege) [inline styles]: Good catch @jfirebaugh!  We've gone through and cleaned up a few others as well.  Thanks!

**Comment by bleege** [comment]
Yay!  Thanks so much @zugaldia for contributing!

**Comment by incanus** [comment]
We should have squashed these commits, rebased, force-pushed, and fast-forwarded `master` the single commit, probably.

**Comment by incanus** [comment]
We should move the changes in `android/java/MapboxGLAndroidSDKTestApp/README.md` over to https://github.com/mapbox/mapbox-gl-native/blob/master/INSTALL.md#android to match the other platforms, so build/dev/testing info is all in one place.

**Comment by incanus** [comment]
Is there a way to run these tests locally with a `make` command like on iOS, Mac, Linux, etc? We should hook that up to something like `make atest` for easier testability.

**Pull request merged**